rust/core-lib/src/plugins/manager.rs

```diff
@@ -145,6 +146,35 @@ impl PluginManager {
         }
     }
 
+    // NOTE: This is a temporary API just for tracing.
+    fn request_trace_rpc_sync(&self, method: &str, params: &Value)
+        -> Vec<Result<Value, xi_rpc::Error>>
+    {
+        let num_global_plugins = self.global_plugins.len();
+        let num_non_global_plugins = self.buffer_plugins.len();
+
+        let num_plugins = num_global_plugins + num_non_global_plugins;
+        let mut gathered_results = Vec::with_capacity(num_plugins);
+
+        for plugin in self.global_plugins.values() {
+            let result = plugin.request_trace_rpc_sync(method, &params);
+            gathered_results.push(result);
+        }
+
+        let mut processed_plugins = HashSet::new();
+
+        for plugin in self.buffer_plugins.values().flat_map(|group| group.values()) {
+            if !processed_plugins.insert(plugin.get_identifier()) {
```

Comment:
I think we discussed this on IRC; this isn't strictly necessary. It's an invariant that a given plugin exists in either `global_plugins` or in **one** `buffer_plugin` collection.

Reply:
Did i implement it wrong?  my intent here is to de-dupe plugins that may be for multiple buffers.  i'm not processing global plugins here i *think*.  Did I miss something?

Reply:
A plugin (which refers to an individual process) can only ever exist in one buffer collection. If multiple buffers are running syntect, for instance, each has its own distinct `PluginRef`, communicating with a distinct process.